G4HepEm/G4HepEmInit/CMakeLists.txt

```diff
@@ -21,25 +21,13 @@ set(G4HEPEMInit_sources
   src/G4HepEmStateInit.cc
 )
 
+g4hepem_add_library(g4HepEmInit SOURCES ${G4HEPEMInit_sources} HEADERS ${G4HEPEMInit_headers} LINK g4HepEmData)
+
+if(BUILD_SHARED_LIBS)
+  target_link_libraries(g4HepEmInit PUBLIC ${Geant4_LIBRARIES})
+endif()
 if(BUILD_STATIC_LIBS)
-  add_library(g4HepEmInit STATIC ${G4HEPEMInit_sources})
-else()
-  add_library(g4HepEmInit SHARED ${G4HEPEMInit_sources})
+  # FIXME: Is Geant4_LIBRARIES really correct for a "mixed" use case with both
+  # shared and static libraries?
+  target_link_libraries(g4HepEmInit-static PUBLIC ${Geant4_LIBRARIES})
```

Comment:
It's not correct in general as `Geant4_LIBRARIES` is a list of the targets to link to, and thus only contains either the shared _or_ static names. Given G4HepEM only uses a subset of the libraries, it might be easier to use the explicit target names, e.g.

```cmake
set(G4HepEmInit_GEANT4_TARGETS
  Geant4::G4global
  Geant4::G4materials
  Geant4::G4particles
  Geant4::G4processes)
list(TRANSFORM G4HepEmInit_GEANT4_TARGETS APPEND "-static" OUTPUT_VARIABLE G4HepEmInit_GEANT4_TARGETS_static)

if(BUILD_SHARED_LIBS)
  target_link_libraries(g4HepEmInit PUBLIC ${G4HepEmInit_GEANT4_TARGETS})
endif()
if(BUILD_STATIC_LIBS)
  target_link_libraries(g4HepEmInit PUBLIC ${G4HepEmInit_GEANT4_TARGETS_static})
endif()
```

Since G4HepEM uses the same naming convention as Geant4 for shared/static target names, the Geant4 targets could also be added to the `LINK` arguments of `g4hepem_add_library` directly as that will partition into shared/static as appropriate.

Reply:
+1 for the explicit list, done